main/global/feho/mapvote.scr

```diff
@@ -0,0 +1,350 @@
+/*
+Map Voting System for MOHAA PN Server
+Created by Copilot for issue #3
+
+This script implements an end-of-round map voting system that allows players
+to vote between the next map in rotation and a random alternative.
+
+Features:
+- Triggers voting 60 seconds before round end
+- Players vote by typing '1' or '2' in chat
+- Prevents duplicate voting per player
+- Announces results and sets nextmap cvar
+- Handles edge cases (ties, no votes, small map lists)
+*/
+
+main:
+    // Initialize the map voting system
+    if (level.mapvote_initialized == 1)
+        end
+    
+    level.mapvote_initialized = 1
+    level.voting_active = 0
+    level.vote_map1_tally = 0
+    level.vote_map2_tally = 0
+    level.vote_option1 = ""
+    level.vote_option2 = ""
+    
+    // Subscribe to player text messages for vote handling
+    event_subscribe "player_textMessage" event_player_textMessage
+    
+    // Start the round timer monitoring thread
+    thread monitor_round_timer
+    
+    // Announce system is loaded (optional debug message)
+    // println "MapVote: System initialized"
+    
+end
+
+monitor_round_timer:
+    // Monitor the round timer and trigger voting when appropriate
+    while (1)
+    {
+        // Check if we're in a timed game mode
+        local.timelimit = getcvar "timelimit"
+        if (local.timelimit == 0 || local.timelimit == "0")
+        {
+            // No time limit, skip voting
+            wait 30
+            goto monitor_round_timer
+        }
+        
+        local.timelimit_seconds = local.timelimit * 60
```

Comment:
[2025-07-23 11:23:57 UTC+1.000] (global/feho/mapvote.scr, 52):
        local.timelimit_seconds = local.timelimit * 60
        local.timelimit_seconds = local.timelimit ^
[2025-07-23 11:23:57 UTC+1.000] ^~^~^ Script Error : binary '*' applied to incompatible types 'string' and 'int'